### PR TITLE
Parse addr

### PR DESCRIPTION
**Pull request by Dspil**

### CODE REVIEW

File: pkg/slayers/scion.go [hunk 1]
```diff
@@ -553,19 +553,39 @@ func scionNextLayerTypeL4(t L4ProtocolType) gopacket.LayerType {
 // from the underlaying layer data. Changing the net.Addr object might lead to inconsistent layer
 // information and thus should be treated read-only. Instead, SetDstAddr should be used to update
 // the destination address.
-// @ trusted
-// @ requires false
-func (s *SCION) DstAddr() (net.Addr, error) {
+// @ requires  acc(&s.DstAddrType) && acc(&s.RawDstAddr)
+// @ requires  s.DstAddrType.Has3Bits()
```
Comment (jcp19) [inline pkg/slayers/scion.go:557]: will we be able to establish this property in the call site?

File: pkg/slayers/scion.go [hunk 2]
```diff
@@ -591,16 +611,33 @@ func (s *SCION) SetSrcAddr(src net.Addr) error {
 	return err
 }
 
-// @ trusted
-// @ requires false
-func parseAddr(addrType AddrType, raw []byte) (net.Addr, error) {
+// @ requires  sl.AbsSlice_Bytes(raw, 0, len(raw))
```
Comment (jcp19) [inline pkg/slayers/scion.go:614]: are you sure we are always able to provide this permission `sl.AbsSlice_Bytes(raw, 0, len(raw))` on every call of this function (and on every call of `SrcAddr` and `DstAddr`)?

File: pkg/slayers/scion.go (hunk 1 above)
Comment (Dspil) [reply pkg/slayers/scion.go:557]: I certainly hope so, because without this we can't prove that there won't be an out of bounds reslicing in parseAddr

**Review by jcp19 — APPROVED** [review]
LGTM, can be merged once the CI succeeds and the errors reported are those fixed in #131

**Pull request merged**